src/main/java/com/synopsys/integration/alert/common/descriptor/config/context/DescriptorActionApi.java

```diff
@@ -41,13 +46,45 @@
 
 public abstract class DescriptorActionApi {
 
-    public abstract void validateConfig(final FieldModel fieldModel, final Map<String, String> fieldErrors);
+    public static final String REQUIRED_FIELD_MISSING = "Required field missing";
+
+    public void validateConfig(final Collection<ConfigField> descriptorFields, final FieldModel fieldModel, final Map<String, String> fieldErrors) {
+        for (final ConfigField field : descriptorFields) {
+            final String fieldKey = field.getKey();
+            final Optional<FieldValueModel> optionalField = fieldModel.getField(fieldKey);
+            if (field.isRequired()) {
+                if (optionalField.isPresent()) {
+                    final FieldValueModel fieldValueModel = optionalField.get();
+                    if (fieldValueModel.hasValues()) {
+                        final boolean valuesAllEmpty = fieldValueModel.getValues().stream().allMatch(value -> StringUtils.isBlank(value));
+                        if (valuesAllEmpty) {
+                            fieldErrors.put(fieldKey, REQUIRED_FIELD_MISSING);
+                        }
+                    } else {
+                        if (!fieldValueModel.isSet()) {
+                            fieldErrors.put(fieldKey, REQUIRED_FIELD_MISSING);
+                        }
+                    }
+                } else {
+                    fieldErrors.put(fieldKey, REQUIRED_FIELD_MISSING);
+                }
+            }
```

Comment:
I believe we can pass the optional FieldValueModel to the ConfigField validate method and have an automatic required test there instead of here. This method I imagine would be more focused on building the error map from the results.